src/show-comments/show-comments.js

```diff
@@ -6,14 +6,24 @@
 import { h } from 'dom-chef'
 import SelectorObserver from 'selector-observer'
 
-const onChange = ({ target, target: { checked: showComments } }) => {
-    const diff = target.closest('section.bb-udiff')
+import './show-comments.css'
+
+export let stateShowComments = true
```

Comment:
There's a bug with the current implementation that stems from this: the fact that you are sharing the same state variable for all of the diff containers.

To reproduce, collapse the comments of one container, then try to collapse the comments of another one. You'll notice that the button doesn't do anything until you click again.

The collapsing state should be encapsulated and isolated to each diff. For that, the original approach worked better. Since you are using a `button` instead of a `input[type="checkbox"]` now, you can't rely on the `checked` property of the DOM node, but that doesn't mean you can't rely on another one. I can recommend `aria-expanded` since it is a standard property which is semantic enough for this purpose.

You could do something like:

```tsx
const onClick = ({ currentTarget, target }) => {
    const ariaExpanded = currentTarget.getAttribute('aria-expanded')
    const isExpanded = ariaExpanded === 'true'
    currentTarget.setAttribute('aria-expanded', !isExpanded)

    // ...
}
```  

And now that you have an attribute on the node to determine its state, you no longer really need the `.__rbb_comments_hidden` classes at all to determine its style.